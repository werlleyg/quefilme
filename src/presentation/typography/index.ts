export * from "./H3";
export * from "./H4";
export * from "./P";
