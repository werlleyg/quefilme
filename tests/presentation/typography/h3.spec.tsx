import { fireEvent, render, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";
import { H3 } from "@/presentation";
import React from "react";

describe("H3 Typography Component", () => {
  it("Should render the H3 component with children", () => {
    const { getByText } = render(<H3>Test Heading</H3>);
    const headingElement = getByText("Test Heading");
    expect(headingElement).toBeInTheDocument();
  });

  it("Should render multiple children correctly", () => {
    const { getByText } = render(
      <H3>
        <span>Child 1</span>
        <span>Child 2</span>
      </H3>,
    );
    expect(getByText("Child 1")).toBeInTheDocument();
    expect(getByText("Child 2")).toBeInTheDocument();
  });

  it("Should apply the correct HTML tag (H3)", () => {
    const { container } = render(<H3>Check Tag</H3>);
    const H3Element = container.querySelector("h3");
    expect(H3Element).toBeInTheDocument();
    expect(H3Element?.tagName).toBe("H3");
  });

  it("Should match the snapshot", () => {
    const { asFragment } = render(<H3>Snapshot Test</H3>);
    expect(asFragment()).toMatchSnapshot();
  });
});
