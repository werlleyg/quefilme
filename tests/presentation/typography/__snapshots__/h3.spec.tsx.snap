// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H3 Typography Component Should match the snapshot 1`] = `
<DocumentFragment>
  <h3
    class="css-16qsii6"
  >
    Snapshot Test
  </h3>
</DocumentFragment>
`;
